hoodie-common/src/main/java/com/uber/hoodie/common/util/SerializationUtils.java

```diff
@@ -16,156 +16,126 @@
 
 package com.uber.hoodie.common.util;
 
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.twitter.chill.EmptyScalaKryoInstantiator;
 import com.uber.hoodie.exception.HoodieSerializationException;
-import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
-import java.io.OutputStream;
 import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicReference;
+
 
 /**
- * (NOTE: Adapted from Apache commons-lang3)
- * This class defines API's to serde an object.
+ * {@link SerializationUtils} class internally uses {@link Kryo} serializer for serializing
+ * / deserializing objects.
  */
 public class SerializationUtils {
+
+  // Caching kryo serializer to avoid creating kryo instance for every serde operation
+  private static final AtomicReference<KryoSerializerInstance> serializerRef =
```

Comment:
is this threadsafe? in case someone runs with more than 1 core?  Consider using ThreadLocal?

Reply:
It is thread safe. See here. http://tutorials.jenkov.com/java-util-concurrent/atomicreference.html

With ThreadLocal we will end up creating more than one copy of it.

Reply:
AtomicReference just provides the ability atomically manipulate the reference itself. I am talking about two  threads using the same reference.. i.e is the KryoSerializerInstance itself threadsafe? 

>> ThreadLocal we will end up creating more than one copy of it.

that's the idea. Have one per write task.

Reply:
We are guaranteeing that single thread is using KryoSerializerInstance by getAndSet() in borrowKryoSerializerInstance(). But I think it makes sense to use ThreadLocal instead. Let me change it.